sysconfig/sysconfig.go

```diff
@@ -35,6 +39,25 @@ type Options struct {
 	// data, i.e. for cloud-init during the initramfs it will be something like
 	// boot.InstallHostWritableDir
 	TargetRootDir string
+
+	// GadgetDir is the path of the mounted gadget snap.
+	GadgetDir string
+}
+
+type FilesystemOnlyApplyOptions struct {
+	// Classic is true when the system in rootdir is a classic system
+	Classic bool
+}
+
+// ConfigcoreFilesystemOnlyApplyImpl is initialized by init() of configcore.
+var ConfigcoreFilesystemOnlyApplyImpl = func(rootDir string, defaults map[string]interface{}, options *FilesystemOnlyApplyOptions) error {
+	panic("configcoreFilesystemOnlyApply not set")
+}
+
+// filesystemOnlyApply applies filesystem modifications under rootDir via
+// configcore.filesystemOnlyApply().
+func ConfigcoreFilesystemOnlyApply(rootDir string, defaults map[string]interface{}, options *FilesystemOnlyApplyOptions) error {
```

Comment:
I would remove the Configcore prexis,  the package name 'sysconfig' has "config" in it as well

Reply:
mmh, actually maybe here this should be called  ApplyFilesystemOnlyDefaults